src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -1226,6 +1226,39 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Dim result As BoundExpression = node
             Dim member As WellKnownMember
 
+            Dim operand = node.Operand
+
+            If operand.Kind = BoundKind.Call AndAlso underlyingTypeTo.IsIntegralType() Then
+                Dim callOperand = DirectCast(operand, BoundCall)
+                Dim target As BoundExpression = Nothing
+                If IsDecimalTruncation(callOperand) Then
+                    ' CInt(Fix(number)) and the like can be simplified to just truncate the number to the integral type
+                    target = callOperand.Arguments(0)
+                ElseIf ReturnsWholeNumberDecimal(callOperand) Then
+                    ' CInt(Math.Floor(number)) and the like can omit rounding the result of Floor, which is already a whole number
+                    target = operand
+                End If
+                If target IsNot Nothing Then
+                    Dim castMember As SpecialMember = SpecialMember.Count
+                    Select Case underlyingTypeTo.SpecialType
+                        Case SpecialType.System_SByte : castMember = SpecialMember.System_Decimal__op_Explicit_ToSByte
+                        Case SpecialType.System_Byte : castMember = SpecialMember.System_Decimal__op_Explicit_ToByte
+                        Case SpecialType.System_Int16 : castMember = SpecialMember.System_Decimal__op_Explicit_ToInt16
+                        Case SpecialType.System_UInt16 : castMember = SpecialMember.System_Decimal__op_Explicit_ToUInt16
+                        Case SpecialType.System_Int32 : castMember = SpecialMember.System_Decimal__op_Explicit_ToInt32
+                        Case SpecialType.System_UInt32 : castMember = SpecialMember.System_Decimal__op_Explicit_ToUInt32
+                        Case SpecialType.System_Int64 : castMember = SpecialMember.System_Decimal__op_Explicit_ToInt64
+                        Case SpecialType.System_UInt64 : castMember = SpecialMember.System_Decimal__op_Explicit_ToUInt64
+                    End Select
+                    If castMember <> SpecialMember.Count Then
+                        Dim castMemberSymbol = DirectCast(ContainingAssembly.GetSpecialTypeMember(castMember), MethodSymbol)
+                        If Not ReportMissingOrBadRuntimeHelper(target, castMember, castMemberSymbol) Then
```

Comment:
>If Not ReportMissingOrBadRuntimeHelper(target, castMember, castMemberSymbol) Then [](http://example.com/codeflow?start=24&length=81)

Is there a chance that existing VB code will be broken (will fail to compile) if this call reports an error? Usually this can be observed in a unit  test by marking specific ```castMember``` as missing (see ```internal void MakeMemberMissing(SpecialMember member)``` API and the way it is used in tests). If compilation for scenario prior to the change was succeeding, it should keep succeeding when these helpers are missing.

Reply:
I would like vbc to take the conventional way instead if this call reports an error, but I do not know how to do.

Reply:
I used `ReportMissingOrBadRuntimeHelper` just because code around there has taken the same way. I do not know when it should be omitted.

Reply:
> I used `ReportMissingOrBadRuntimeHelper` just because code around there has taken the same way. I do not know when it should be avoided.

I suggest considering using the following helper:
```
        Private Function TryGetSpecialMember(Of T As Symbol)(<Out> ByRef result As T,
                                                       memberId As SpecialMember,
                                                       syntax As SyntaxNode,
                                                       Optional isOptional As Boolean = False) As Boolean
``` 

Please add tests for compiling the affected scenarios with the newly used APIs marked as missing.

Reply:
> I suggest considering using the following helper:

This seems to help me a lot.

> Please add tests for compiling the affected scenarios with the newly used APIs marked as missing.

Sorry I have no idea. I do not know how I can intentionally disable decimal-to-integral-types casts.

Reply:
>> Please add tests for compiling the affected scenarios with the newly used APIs marked as missing.

> Sorry I have no idea. I do not know how I can intentionally disable decimal-to-integral-types casts.

Special members can be marked as missing for test purposes. Compilation object has an API called "MakeMemberMissing". There should be many examples of its usage in our tests.